### PR TITLE
feat(rest) : End point for export vendor spreadsheet

### PR DESCRIPTION
**Pull request by keerthi-bl**
[//]: # (Copyright Bosch.IO GmbH 2020)
[//]: # (This program and the accompanying materials are made)
[//]: # (available under the terms of the Eclipse Public License 2.0)
[//]: # (which is available at https://www.eclipse.org/legal/epl-2.0/)
[//]: # (SPDX-License-Identifier: EPL-2.0)

> Please provide a summary of your changes here.
> * Which issue is this pull request belonging to and how is it solving it? (*Refer to issue here*)
Issue:  #2124 

> * Did you add or update any new dependencies that are required for your change?
No 

It is a rest end point. It can be tested using Postman client, as follows.

URL : http://localhost:8080/resource/api/vendors/exportVendorDeatails

Click on Send and Download and check the downloaded excel file. Attached sample excel file here.

Sample response : "OK"

image

> Have you implemented any additional tests?

### Checklist
Must:
- [ ] All related issues are referenced in commit messages and in PR

### CODE REVIEW

File: backend/src/src-vendors/src/main/java/org/eclipse/sw360/vendors/VendorDatabaseHandler.java [hunk 1]
```diff
@@ -238,4 +242,13 @@ private boolean isVendorUnderModeration(String vendorId) throws TException {
         return sourceModerationRequests.stream().anyMatch(CommonUtils::isInProgressOrPending);
     }
 
+    public ByteBuffer getVendorReportDataStream(List<Vendor> vendorList) throws TException{
+        try {
+            InputStream stream = new VendorExporter().makeExcelExport(vendorList);
+            return ByteBuffer.wrap(IOUtils.toByteArray(stream));
+        }catch (Exception e) {
```
Comment (hoangnt2) [inline backend/src/src-vendors/src/main/java/org/eclipse/sw360/vendors/VendorDatabaseHandler.java:249]: add space after }

File: rest/resource-server/src/test/java/org/eclipse/sw360/rest/resourceserver/restdocs/VendorSpecTest.java [hunk 2]
```diff
@@ -146,4 +149,14 @@ public void should_document_create_vendor() throws Exception {
                                 subsectionWithPath("_links").description("<<resources-index-links,Links>> to other resources")
                         )));
     }
+
+    @Test
+    public void should_document_get_export_vendor() throws Exception{
+        String accessToken = TestHelper.getAccessToken(mockMvc, testUserId, testUserPassword);
+        mockMvc.perform(get("/api/vendors/exportVendorDeatails")
```
Comment (hoangnt2) [inline rest/resource-server/src/test/java/org/eclipse/sw360/rest/resourceserver/restdocs/VendorSpecTest.java:156]: typo **exportVendorDeatails** -> **exportVendorDetails**

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/vendor/VendorController.java [hunk 3]
```diff
@@ -112,4 +118,22 @@ public RepositoryLinksResource process(RepositoryLinksResource resource) {
     private HalResource<Vendor> createHalVendor(Vendor sw360Vendor) {
         return new HalResource<>(sw360Vendor);
     }
+
+    @PreAuthorize("hasAuthority('WRITE')")
+    @GetMapping(value = VENDORS_URL + "/exportVendorDeatails")
```
Comment (hoangnt2) [inline rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/vendor/VendorController.java:123]: typo **exportVendorDeatails** -> **exportVendorDetails**

**Review by hoangnt2 — CHANGES_REQUESTED** [review]

**Comment by rudra-superrr** [comment]
Testing was successful.

**Comment by hoangnt2** [comment]
Code look good

**Pull request merged**